main.go

```diff
@@ -1,639 +1,60 @@
-				sourceTestOutputDir:   opts.XcresultPath,
-				targetAddonPath:       addonResultPath,
-				targetAddonBundleName: opts.Scheme,
-			}); err != nil {
-				log.Warnf("Failed to export test results, error: %s", err)
-			}
-		}
-	}
-
-	// export xcodebuild build log
-	if opts.XcodebuildBuildLog != "" {
-		pth, err := saveRawOutputToLogFile(opts.XcodebuildBuildLog)
-		if err != nil {
-			log.Warnf("Failed to save the Raw Output, err: %s", err)
-		}
-
-		deployPth := filepath.Join(opts.DeployDir, "xcodebuild_build.log")
-		if err := command.CopyFile(pth, deployPth); err != nil {
-			return fmt.Errorf("failed to copy xcodebuild output log file from (%s) to (%s), error: %s", pth, deployPth, err)
-		}
-
-		if err := cmd.ExportEnvironmentWithEnvman("BITRISE_XCODEBUILD_BUILD_LOG_PATH", deployPth); err != nil {
-			log.Warnf("Failed to export: BITRISE_XCODEBUILD_BUILD_LOG_PATH, error: %s", err)
-		}
-	}
-
-	// export xcodebuild test log
-	if opts.XcodebuildTestLog != "" {
-		pth, err := saveRawOutputToLogFile(opts.XcodebuildTestLog)
-		if err != nil {
-			log.Warnf("Failed to save the Raw Output, error: %s", err)
-		}
-
-		deployPth := filepath.Join(opts.DeployDir, "xcodebuild_test.log")
-		if err := command.CopyFile(pth, deployPth); err != nil {
-			return fmt.Errorf("failed to copy xcodebuild output log file from (%s) to (%s), error: %s", pth, deployPth, err)
-		}
-
-		if err := cmd.ExportEnvironmentWithEnvman("BITRISE_XCODEBUILD_TEST_LOG_PATH", deployPth); err != nil {
-			log.Warnf("Failed to export: BITRISE_XCODEBUILD_TEST_LOG_PATH, error: %s", err)
-		}
-	}
-
-	// export simulator diagnostics log
-	if opts.SimulatorDiagnosticsPath != "" {
-		diagnosticsName, err := simulatorDiagnosticsName()
-		if err != nil {
-			return err
-		}
-
-		outputPath := filepath.Join(opts.DeployDir, diagnosticsName)
-		if err := ziputil.ZipDir(opts.SimulatorDiagnosticsPath, outputPath, true); err != nil {
-			return fmt.Errorf("failed to compress simulator diagnostics result: %v", err)
-		}
-	}
-
-	// export UITest artifacts
-	if opts.ExportUITestArtifacts && opts.XcresultPath != "" {
-		// The test result bundle (xcresult) structure changed in Xcode 11:
-		// it does not contains TestSummaries.plist nor Attachments directly.
-		fmt.Println()
-		log.Infof("Exporting attachments")
-
-		testSummariesPath, attachementDir, err := getSummariesAndAttachmentPath(opts.XcresultPath)
-		if err != nil {
-			log.Warnf("Failed to export UI test artifacts, error: %s", err)
-		}
-
-		if err := saveAttachments(opts.Scheme, testSummariesPath, attachementDir); err != nil {
-			log.Warnf("Failed to export UI test artifacts, error: %s", err)
-		}
-	}
-
-	return nil
-}
-
-func run() error {
-	step := NewStep()
-	config, err := step.ProcessConfig()
-	if err != nil {
-		return err
-	}
-
-	if err := step.InstallDeps(config.OutputTool == xcprettyTool); err != nil {
-		config.OutputTool = xcodebuildTool
-	}
-
-	res, runErr := step.Run(config)
-
-	opts := ExportOpts{
+	opts := step.ExportOpts{
```

Comment:
Since it consists of the Run result:

```
	XcresultPath             string
	XcodebuildBuildLog       string
	XcodebuildTestLog        string
	SimulatorDiagnosticsPath string
````

and some of the Configs:

```
	Scheme                string
	DeployDir             string
	ExportUITestArtifacts bool
```

(TestFailed is calculated)

Reply:
What about `func (s XcodeTestRunner) Export(res RunResult, opts ExportOpts) error {` ?

Reply:
Also Run gets all the Configs, while not all of its properties are used. 
What do you think of introducing a new input model for Run, I prefer when functions are receiving only what they are using.

Reply:
I think of this method as a "generic" one that can contain no other code than calling the parse install run export methods. I think it is the "run"'s responsiblity to merge its result with the config and provide the output for the export.

Reply:
fixed in https://github.com/bitrise-steplib/steps-xcode-test/pull/181/commits/9ee746d5daa4b4c48aa322d6ce4757df10559364